src/parser/flags.ts

```diff
@@ -35,13 +35,21 @@ export function boolean<T = boolean>(
   } as BooleanFlag<T>
 }
 
-export const integer = build({
-  parse: async input => {
-    if (!/^-?\d+$/.test(input))
-      throw new Error(`Expected an integer but received: ${input}`)
-    return Number.parseInt(input, 10)
-  },
-})
+export const integer = (opts: { min?: bigint; max?: bigint } & Partial<OptionFlag<number>> = {}): OptionFlag<number | undefined> => {
+  return build<number>({
+    ...opts,
+    parse: async input => {
+      if (!/^-?\d+$/.test(input))
+        throw new Error(`Expected an integer but received: ${input}`)
+      const num = Number.parseInt(input, 10)
+      if (typeof opts.min === 'number' && num < opts.min)
+        throw new Error(`Expected an integer greater than or equal to ${opts.min} but received: ${input}`)
+      if (typeof opts.max === 'number' && num > opts.max)
+        throw new Error(`Expected an integer less than or equal to ${opts.max} but received: ${input}`)
+      return num
```

Comment:
If I'm reading this correctly if `opts.min` or `opts.max` aren't typeof number they'll just be ignored and we'll return w/e the user passed in?

Should probs add an error if they're something other than typeof number (yeah I know TS should avoid this but what if someone doesn't have TS setup on their editor?).

Reply:
decision from synchronous discussion:

typeof was handling the optional properties, but excluding `undefined` would allow runtime errors on min/max for non-number types that Rod wants.

Reply:
Just to clarify, it's not that I _want_ this to error but the developer should be made aware if they are using non-numerical values for `min`/`max` that they can't do that. That wouldn't have happened with the previous `typeof min === 'number'` check. It would have just returned w/e the user passed in without comparing it to a min, same for max.

Would be ideal if these could be compile time errors and that's why we use TS but even that won't catch all scenarios.